File: LivrusVirtus/KDevelop/wxWin/ODBC/Motif/plivrus/ConIdiomas.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConIdiomas.h"

BEGIN_EVENT_TABLE(CfrmConIdiomas, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConIdiomas::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConIdiomas::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConIdiomas::OnBtnFechar)
  EVT_CLOSE(CfrmConIdiomas::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConIdiomas::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConIdiomas::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConIdiomas::CfrmConIdiomas(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConIdiomas::~CfrmConIdiomas() { }

void CfrmConIdiomas::IniciaControles() {
 this->SetTitle(STR_CONIDIOMA);
 this->labDescricao->SetLabel(wxT("Idioma:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("Idioma"));
}

void CfrmConIdiomas::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_IDIOMAS);
  event.Skip();
}

void CfrmConIdiomas::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConIdiomas::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConIdiomas::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Idioma('S',wxT(""), edtDescricao->GetValue()), STR_IDINENC, 2);
  event.Skip();
}

void CfrmConIdiomas::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConIdiomas::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/ConIdiomas.pas
unit ConIdiomas;

interface

uses
  System.Drawing, wx, Consultas;

type
  TfrmConIdiomas = class(TfrmConsultas)
  private
    procedure OnClose(sender: System.Object; e: wx.Event);
    procedure OnBtnFechar(sender: System.Object; e: wx.Event);
    procedure OnBtnLimpar(sender: System.Object; e: wx.Event);
    procedure OnBtnPesquisar(sender: System.Object; e: wx.Event);
    procedure OnGridSelectCell(sender: System.Object; e: wx.Event);
    procedure OnGridLabelLeftClick(sender: System.Object; e: wx.Event);
  public
    constructor Create(parent: wx.Window);
  end;

var
  frmConIdiomas: TfrmConIdiomas;

  [assembly: RuntimeRequiredAttribute(TypeOf(TfrmConIdiomas))]

implementation

uses Resource, ConsultasSQL;

constructor TfrmConIdiomas.Create(parent: wx.Window);
begin
  inherited Create(parent, 'Idioma:');

  Title := Res.STR_CONIDIOMA;
  //labDescricao.Label := 'Idioma:';
  gridConsulta.SetColLabelValue(1, 'Idioma');

  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, OnBtnPesquisar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, OnBtnLimpar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, OnBtnFechar );
  EVT_CLOSE( OnClose );
  EVT_GRID_SELECT_CELL( OnGridSelectCell );
  EVT_GRID_LABEL_LEFT_CLICK( OnGridLabelLeftClick );
end;

procedure TfrmConIdiomas.OnClose(sender: System.Object; e: wx.Event);
begin
  Fecha(Res.MNU_CON_IDIOMAS);
	e.Skip();
end;

procedure TfrmConIdiomas.OnBtnFechar(sender: System.Object; e: wx.Event);
begin
  Fecha(0);
  e.Skip();
end;

procedure TfrmConIdiomas.OnBtnLimpar(sender: System.Object; e: wx.Event);
begin
  LimparDados();
  e.Skip();
end;

procedure TfrmConIdiomas.OnBtnPesquisar(sender: System.Object; e: wx.Event);
begin
  PesquisarDados(ConsSQL.Idioma('S','', edtDescricao.Value),
    Res.STR_IDINENC, 2);
  e.Skip();
end;

procedure TfrmConIdiomas.OnGridSelectCell(sender: System.Object; e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

procedure TfrmConIdiomas.OnGridLabelLeftClick(sender: System.Object;
  e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

end.
